flink-cdc-e2e-tests/flink-cdc-pipeline-e2e-tests/src/test/java/org/apache/flink/cdc/pipeline/tests/MaxComputeE2eITCase.java

```diff
@@ -0,0 +1,193 @@
+import org.apache.flink.cdc.common.test.utils.TestUtils;
+import org.apache.flink.cdc.connectors.maxcompute.options.MaxComputeOptions;
+import org.apache.flink.cdc.connectors.maxcompute.utils.MaxComputeUtils;
+import org.apache.flink.cdc.pipeline.tests.utils.PipelineTestEnvironment;
+import org.apache.flink.client.program.rest.RestClusterClient;
+import org.apache.flink.runtime.client.JobStatusMessage;
+import org.apache.flink.table.api.ValidationException;
+
+import com.aliyun.odps.Instance;
+import com.aliyun.odps.Odps;
+import com.aliyun.odps.data.Record;
+import com.aliyun.odps.task.SQLTask;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.InetAddress;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/** End-to-end tests for maxcompute cdc pipeline job. */
+public class MaxComputeE2eITCase extends PipelineTestEnvironment {
+    private static final Logger LOG = LoggerFactory.getLogger(MaxComputeE2eITCase.class);
+
+    public static final DockerImageName MAXCOMPUTE_IMAGE =
+            DockerImageName.parse("maxcompute/maxcompute-emulator:v0.0.4");
+
+    @ClassRule
+    public static GenericContainer<?> maxcompute =
+            new GenericContainer<>(MAXCOMPUTE_IMAGE)
+                    .withExposedPorts(8080)
+                    .waitingFor(
+                            Wait.forLogMessage(".*Started MaxcomputeEmulatorApplication.*\\n", 1))
+                    .withLogConsumer(frame -> System.out.print(frame.getUtf8String()));
+
+    public final MaxComputeOptions testOptions =
+            MaxComputeOptions.builder("ak", "sk", getEndpoint(), "mocked_mc")
+                    .withTunnelEndpoint(getEndpoint())
+                    .build();
+
+    @Test
+    public void testSingleSplitSingleTable() throws Exception {
+        startTest("SINGLE_SPLIT_SINGLE_TABLE");
+        Instance instance =
+                SQLTask.run(
+                        MaxComputeUtils.getOdps(testOptions),
+                        "select * from table1 order by col1;");
+        instance.waitForSuccess();
+        List<Record> result = SQLTask.getResult(instance);
+        System.out.println(result);
+        Assert.assertEquals(2, result.size());
+        // 2,x
+        Assert.assertEquals("2", result.get(0).get(0));
+        Assert.assertEquals("x", result.get(0).get(1));
+        // 3, NULL (MaxCompute Emulator use 'NULL' instead of null)
+        Assert.assertEquals("3", result.get(1).get(0));
+        Assert.assertEquals("NULL", result.get(1).get(1));
```

Comment:
What about add more verification of insert/update records and schema change after snapshot reading.